Sources_v3/Controllers/EntityDatabaseObserver_Tests.swift

```diff
@@ -131,6 +131,122 @@ class EntityDatabaseObserver_Tests: XCTestCase {
         AssertAsync.willBeEqual(listener1Changes, expectedChanges)
         AssertAsync.willBeEqual(listener2Changes, expectedChanges)
     }
+    
+    func test_onFieldChange_forwardsCreateFieldChange() throws {
+        let testItem: TestItem = .unique
+        fetchRequest.predicate = NSPredicate(format: "testId == %@", testItem.id)
+        
+        // Add new field listener
+        var lastChange: EntityChange<String?>?
+        observer.onFieldChange(\.value) {
+            lastChange = $0
+        }
+        
+        // Start observing
+        try observer.startObserving()
+        
+        // Insert a new entity matching the predicate
+        let insertError = try await {
+            database.write({
+                let new = TestManagedObject(context: $0 as! NSManagedObjectContext)
+                new.testValue = testItem.value
+                new.testId = testItem.id
+            }, completion: $0)
+        }
+        
+        // Assert insertation happened without any error
+        XCTAssertNil(insertError)
```

Comment:
Can you change this to `database.writeSynchronously { }` so we can remove this boilerplate from tests?

Reply:
I see `TestDatabaseContainer` and `DatabaseContainerMock` in the codebase. Shouldn't we have only one type and update all the tests?

Reply:
Right. Would you mind doing it?

Reply:
sure. Will do in a separate PR

Reply:
done